src/AppBundle/Helper/CardFilter/LabelNameCardFilter.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+
+namespace AppBundle\Helper\CardFilter;
+
+
+use AppBundle\Helper\TrelloClient\Model\Card;
+use AppBundle\Helper\TrelloClient\Model\Label;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+
+class LabelNameCardFilter extends AbstractCardFilter
+{
+	/**
+	 * @var Label
+	 */
+	private $targetLabel;
+
+	/**
+	 * Initialization of the filter should be implemented in this function. This may include asking the user for input.
+	 * If setup needs to be cancelled, then an exception can be thrown for the filter not to be created
+	 *
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @param string $boardId
+	 *
+	 * @return CardFilterInterface
+	 * @throws \Exception
+	 */
+	static public function setUp(InputInterface $input, OutputInterface $output, $boardId)
+	{
+		$trelloClient = self::getContainer()->get('trello_client');
+
+		$boardLabels = $trelloClient->getBoardLabels($boardId);
+		$options = [];
+		$count = 1;
+		foreach($boardLabels as $boardLabel ){
+			$labelName = $boardLabel->getName();
+			if ( isset($options[ $boardLabel->getName() ]) && ( isset( $labelName ) && !empty( $boardLabel->getName()) ) )
+			{
+				$options[$count.' '.$boardLabel->getName() ] = $boardLabel;
+			}
+			elseif( isset( $labelName ) && !empty( $boardLabel->getName()) )
+			{
+				$options[$count.' '.$boardLabel->getName()] = $boardLabel;
+			}
+			$count ++;
+			 //$options[$boardLabel->getId()] = $boardLabel;
+		}
+		$filterOptionQuestion = new ChoiceQuestion(
+			'Select Label to filter by:',
+			array_merge(array_keys($options), ['<- Back'])
+		);
+		$questionHelper = new QuestionHelper();
```

Comment:
The display for the setup renders as follows:

```
Configure Filter: Has Label
Select Label to filter by:
  [0] 1 Alpha
  [1] 3 Alpha
  [2] 4 Bravo
  [3] <- Back
 >
```

You can rely on the `ChoiceQuestion` to do the option numbering for you